src/tasks/mod.rs

```diff
@@ -472,33 +467,84 @@ where
 /// add the consensus task
 /// # Panics
 /// Is unable to panic. This section here is just to satisfy clippy
-pub async fn add_consensus_task(
+pub async fn add_consensus_task<
+    TYPES: NodeType<ConsensusType = SequencingConsensus, Time = ViewNumber>,
+    I: NodeImplementation<
+        TYPES,
+        Leaf = SequencingLeaf<TYPES>,
+        ConsensusMessage = SequencingMessage<TYPES, I>,
+    >,
+>(
     task_runner: TaskRunner,
-    event_stream: ChannelStream<GlobalEvent>,
-) -> TaskRunner {
-    // build the consensus task
-    let consensus_state = ConsensusTaskState {};
+    event_stream: ChannelStream<SequencingHotShotEvent<TYPES, I>>,
+    hotshot: SystemContext<TYPES::ConsensusType, TYPES, I>,
+) -> TaskRunner
+where
+    I::Exchanges: SequencingExchangesType<TYPES, Message<TYPES, I>>,
+    SequencingQuorumEx<TYPES, I>: ConsensusExchange<
+        TYPES,
+        Message<TYPES, I>,
+        Proposal = QuorumProposal<TYPES, SequencingLeaf<TYPES>>,
+        Certificate = QuorumCertificate<TYPES, SequencingLeaf<TYPES>>,
+        Commitment = SequencingLeaf<TYPES>,
+    >,
+    CommitteeEx<TYPES, I>: ConsensusExchange<
+        TYPES,
+        Message<TYPES, I>,
+        Certificate = DACertificate<TYPES>,
+        Commitment = TYPES::BlockType,
+    >,
+{
+    let consensus = hotshot.get_consensus();
+    let c_api: HotShotSequencingConsensusApi<TYPES, I> = HotShotSequencingConsensusApi {
```

Comment:
Do we have any plans to nuke this consensus api? I remember discussing this a while back, don't remember the outcome of that conversation.

Reply:
Hmm I'm not sure, I don't remember the discussion from before.  Here it's just needed to get the exchanges from consensus object.  I'd like to discuss this part since it looks like every tasks needs something like this passed in.